packages/ddex/ingester/indexer/indexer.go

```diff
@@ -1,62 +1,266 @@
 package indexer
 
 import (
+	"archive/zip"
 	"context"
+	"fmt"
 	"ingester/common"
-	"ingester/constants"
+	"io"
 	"log"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"strings"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/oklog/ulid/v2"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
 	"go.mongodb.org/mongo-driver/mongo"
 )
 
-func Run(ctx context.Context) {
+type Indexer struct {
+	s3Downloader  *s3manager.Downloader
+	s3Uploader    *s3manager.Uploader
+	mongoClient   *mongo.Client
+	rawBucket     string
+	indexedBucket string
+	indexedColl   *mongo.Collection
+	ctx           context.Context
+	logger        *slog.Logger
+}
+
+// RunNewIndexer starts the indexer service, which listens for new uploads in the Mongo "uploads" collection and processes them.
+func RunNewIndexer(ctx context.Context) {
+	_, s3Session := common.InitS3Client()
 	mongoClient := common.InitMongoClient(ctx)
 	defer mongoClient.Disconnect(ctx)
+	indexedColl := mongoClient.Database("ddex").Collection("indexed")
+
+	i := &Indexer{
+		s3Downloader:  s3manager.NewDownloader(s3Session),
+		s3Uploader:    s3manager.NewUploader(s3Session),
+		mongoClient:   mongoClient,
+		rawBucket:     os.Getenv("AWS_BUCKET_RAW"),
```

Comment:
can we do a `common.MustGetenv` for these

Reply:
done ✅